node/src/components/consensus/era_supervisor.rs

```diff
@@ -312,13 +311,24 @@ where
         // Activate the era if this node was already running when the era began, it is still
         // ongoing based on its minimum duration, and we are one of the validators.
         let our_id = &self.public_signing_key;
-        let should_activate = if !validators.contains_key(&our_id) {
+        let should_activate = if self.current_era > era_id {
+            trace!(
+                era = era_id.value(),
+                current_era = self.current_era.value(),
+                "not voting; initializing past era"
+            );
+            false
```

Comment:
I thought eras are always initialized in order – from the oldest one to the latest one – so `self.current_era` should never be bigger than `era_id`.

Reply:
But on initialization we set `current_era`, before we initialize the past eras: https://github.com/CasperLabs/casper-node/blob/42ec609300d6bf934fe86fdeed5a0b64c7c967e0/node/src/reactor/validator.rs#L469

Reply:
Thanks, I did not remember that.